tests/sdn/persistentips.go

```diff
@@ -0,0 +1,153 @@
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package sdn
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	nadv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	kvirtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/framework/checks"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libmigration"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = SIGDescribe("Persistent IPs", func() {
+
+	Context("with a VirtualMachine configured with a allowPersistentIPs nad", func() {
+		var (
+			networkInterfaceName = "multus"
+			vm                   *kvirtv1.VirtualMachine
+			vmi                  *kvirtv1.VirtualMachineInstance
+			nad                  *nadv1.NetworkAttachmentDefinition
+		)
+		BeforeEach(func() {
+			Expect(checks.HasFeature(virtconfig.PersistentIPsGate)).To(BeTrue(), "should have persistent ips feature gate enabled")
+
+			nad = &nadv1.NetworkAttachmentDefinition{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "persistentips",
+					Namespace: testsuite.GetTestNamespace(nil),
+				},
+				Spec: nadv1.NetworkAttachmentDefinitionSpec{
+					Config: fmt.Sprintf(`
+{
+        "cniVersion": "0.3.0",
+        "name": "persistentips",
+        "type": "ovn-k8s-cni-overlay",
+        "topology": "layer2",
+        "subnets": "10.100.200.0/24",
+        "netAttachDefName": "%s/persistentips",
+        "allowPersistentIPs": true
+}
+`, testsuite.GetTestNamespace(nil)),
+				},
+			}
+			By("Create NetworkAttachmentDefinition")
+			_, err := kubevirt.Client().NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(testsuite.GetTestNamespace(nil)).Create(context.TODO(), nad, metav1.CreateOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			vm = libvmi.NewVirtualMachine(libvmifact.NewAlpine(
+				libvmi.WithNamespace(testsuite.GetTestNamespace(nil)),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(networkInterfaceName)),
+				libvmi.WithNetwork(libvmi.MultusNetwork(networkInterfaceName, "persistentips")),
+			), libvmi.WithRunning())
+
+			By("Creating VM using the nad")
+			vm, err = kubevirt.Client().VirtualMachine(vm.Namespace).Create(context.Background(), vm, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Wait for VMI to be created")
+			Eventually(matcher.ThisVMIWith(vm.Namespace, vm.Name)).
+				WithPolling(time.Second).
+				WithTimeout(2 * time.Minute).
+				ShouldNot(BeNil())
+			vmi, err = matcher.ThisVMIWith(vm.Namespace, vm.Name)()
```

Comment:
@qinqon This looks like not what the API is for, can we do something like this instead:
```
                        vmi := libvmifact.NewAlpine(
				libvmi.WithNamespace(testsuite.GetTestNamespace(nil)),
				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(networkInterfaceName)),
				libvmi.WithNetwork(libvmi.MultusNetwork(networkInterfaceName, "persistentips")),
			)

                        vm := libvmi.NewVirtualMachine(vmi ,  libvmi.WithRunning())
```
With that you will have the VMI already so you don't need to call the matcher for that.